README.md

```diff
@@ -1,8 +1,246 @@
+to run `dartfmt -w .`. But, some projects may want to exclude certain files that
+would otherwise be formatted by this command. Or, some projects may want to use
+`pub run dart_style:format` instead of `dartfmt`. Currently, there is no
+project-level configuration supported by the formatter, so these sorts of things
+just have to be documented in a `README.md` or `CONTRIBUTING.md`.
+
+With `dart_dev`, this can be accomplished like so:
+
+```dart
+// tool/dev.dart
+import 'package:dart_dev/dart_dev.dart';
+
+final config = {
+  'format': FormatTool()
+    ..exclude = [Glob('lib/src/**.g.dart')]
+    ..formatter = Formatter.dartStyle,
+};
+```
+
+```bash
+$ ddev format
+[INFO] Running subprocess:
+pub run dart_style:format -w <3 paths>
+--------------------------------------
+Unchanged ./lib/foo.dart
+Unchanged ./lib/src/bar.dart
+Formatted ./lib/src/baz.dart
+```
+
+## Extending/Composing Functionality
+
+Using existing tooling provided by (or conventionalized by) the Dart community
+should always be the goal, but the reality is that there are gaps. Certain use
+cases can be made more convenient and new use cases may arise.
+
+Consider test running as an example. For simple projects, `pub run test` is
+sufficient. In fact, the test package supports a huge amount of project-level
+configuration via `dart_test.yaml`, which means that for projects that are
+properly configured, `pub run test` just works.
+
+Unfortunately, at this time, projects that rely on builders must run tests via
+`pub run build_runner test`. Based on the project, you would need to know which
+test command should be run.
+
+With `dart_dev`, the `TestTool` handles this automatically by checking the
+project's `pubspec.yaml` for a dependency on `build_test`. If present, tests
+will be run via `pub run build_runner test`, otherwise it falls back to the
+default of `pub run test`.
+
+```bash
+# In a project without a `build_test` dependency:
+$ ddev test
+[INFO] Running subprocess:
+pub run test
+----------------------------
+00:01 +75: All tests passed!
+
+
+# In a project with a `build_test` dependency:
+$ ddev test
+[INFO] Running subprocess:
+pub run build_runner test
+----------------------------
+[INFO] Generating build script completed, took 425ms
+[INFO] Creating build script snapshot... completed, took 13.6s
+[INFO] Building new asset graph completed, took 960ms
+[INFO] Checking for unexpected pre-existing outputs. completed, took 1ms
+[INFO] Running build completed, took 12.4s
+[INFO] Caching finalized dependency graph completed, took 71ms
+[INFO] Creating merged output dir `/var/folders/vb/k8ccjw095q16jrwktw31ctmm0000gn/T/build_runner_testBkm6gS/` completed, took 260ms
+[INFO] Writing asset manifest completed, took 3ms
+[INFO] Succeeded after 12.8s with 1276 outputs (2525 actions)
+Running tests...
+
+00:00 +75: All tests passed!
+```
+
+Additionally, `TestTool` automatically applies [`--build-filter`][build-filter]
```

Comment:
What do you think about including the need to specify a test file path for the `--build-filter` to be applied?